packages/react/src/components/CodeSnippet/CodeSnippet-story.js

```diff
@@ -1,135 +1,321 @@
+        <br />
+        {story()}
+      </div>
+    ),
+  ],
   parameters: {
     docs: {
       page: mdx,
     },
   },
 };
 
-const props = () => ({
-  type: select(
-    'Type (type)',
-    {
-      inline: 'inline',
-      'single line': 'single',
-      'multiple line': 'multi',
-    },
-    'inline'
-  ),
-  disabled: boolean('Disabled (disabled)', false),
-  light: boolean('Light variant (light)', false),
-  feedback: text('Feedback text', 'Copied to clipboard'),
-  showMoreText: text('Text for "show more" button', 'Show more'),
-  showLessText: text('Text for "show less" button', 'Show less'),
-  hideCopyButton: boolean('Hide copy button (hideCopyButton)', false),
+const typeOptions = {
+  inline: { inline: 'inline' },
+  single: { single: 'single' },
+  multi: { multi: 'multi' },
+  skeleton: { single: 'single', multi: 'multi' },
+  playground: {
+    inline: 'inline',
+    single: 'single',
+    multi: 'multi',
+  },
+};
+
+const typeDefaults = {
+  inline: 'inline',
+  single: 'single',
+  multi: 'multi',
+  skeleton: 'single',
+  playground: 'inline',
+};
+
+const childrenOptions = {
+  inline: 'node -v',
+  single:
+    'yarn add carbon-components@latest carbon-components-react@latest @carbon/icons-react@latest carbon-icons@latest',
+  multi: 'The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.\n'.repeat(
+    60
+  ),
+  skeleton: undefined,
+  playground: 'Playground CodeSnippet',
+};
+
+const ariaLabelProp = () => ({
+  ariaLabel: text(
+    'ariaLabel: Specify a label to be read by screen readers',
+    undefined
+  ),
+});
+const classNameProp = () => ({
+  className: text(
+    'className: Specify an optional className to be applied to the container node',
+    undefined
+  ),
+});
+const childrenProp = (type) => ({
+  children: text(
+    'children: Provide the content of your CodeSnippet as a string',
+    childrenOptions[type || 'playground']
+  ),
+});
+const copyButtonDescriptionProp = () => ({
+  copyButtonDescription: text(
+    'copyButtonDescription: Specify the description for the Copy Button',
+    undefined
+  ),
+});
+const disabledProp = () => ({
+  disabled: boolean(
+    'disabled: Specify whether or not the CodeSnippet should be disabled',
+    false
+  ),
+});
+const feedbackProp = () => ({
+  feedback: text(
+    'feedback: Specify the string displayed when the snippet is copied',
+    undefined
+  ),
+});
+const hideCopyButtonProp = () => ({
+  hideCopyButton: boolean(
+    'hideCopyButton: Specify whether or not a copy button should be used/rendered',
+    false
+  ),
+});
+const lightProp = () => ({
+  light: boolean(
+    'light: Specify whether you are using the light variant of the Code Snippet',
+    false
+  ),
+});
+const maxClosedNumberOfRowsProp = () => ({
+  maxClosedNumberOfRows: number(
+    'maxClosedNumberOfRows: Specify the maximum number of rows to be shown when in closed view',
+    15,
+    { min: 1 }
```

Comment:
From @dakahn
It is worth noting is I can select negative rows in the playground addons panel for this.
The negative numbers aren't a blocker, just worth mentioning in case someone with some Storybook experience knew a quick fix.

Agreed, and sounds good.

On the one hand we may want Storybook to be user friendly and only allow what the user should enter.
Towards this I added 'min' entry, and would like to add 'scale' and maybe coded limiters...

On the other hand we may want Storybook to accept anything the user may pass to the Component.
This way we can dup problems, and/or test that bad values are handled properly.

I see both sides, I'm torn (but lean towards leaving it open), but will do whatever Carbon wants.

Reply:
Agreed, and sounds good.

On the one hand we may want Storybook to be user friendly and only allow what the user should enter.
Towards this I added 'min' entry, and would like to add 'scale' and maybe coded limiters...

On the other hand we may want Storybook to accept anything the user may pass to the Component.
This way we can dup problems, and/or test that bad values are handled properly.

I see both sides, I'm torn (but lean towards leaving it open), but will do whatever Carbon wants.